cmd/opensearch-csv-exporter/export.go

```diff
@@ -185,54 +184,155 @@ func search(ctx context.Context, config OpensearchConfig, client *elasticsearch.
 	if err != nil {
 		return "", 0, fmt.Errorf("failed to search: %w", err)
 	}
-	defer response.Body.Close()
-	return decode(response, csv)
+	if response.IsError() {
+		return "", 0, fmt.Errorf("failed to decode, response is error: %s", response)
+	}
+	return decode(response.Body, csv)
 }
 
+var size = 10000
+
 func scroll(ctx context.Context, client *elasticsearch.Client, csv *CSV, body io.Reader) (string, error) {
 	s := client.Scroll
 	response, err := s(s.WithContext(ctx), s.WithBody(body), s.WithScroll(time.Minute))
 	if err != nil {
 		return "", fmt.Errorf("failed to scroll: %w", err)
 	}
-	defer response.Body.Close()
-	scrollId, _, err := decode(response, csv)
+	if response.IsError() {
```

Comment:
If a this is true, does the response still contain a body that should be closed?

Reply:
You are correct. examples and implementation of Do suggests it needs to be closed even if IsError is true.